CompanyEmployees.Application/Validators/CheckInAttendanceCommandValidator.cs

```diff
@@ -0,0 +1,37 @@
+using CompanyEmployees.Application.Commands;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace CompanyEmployees.Application.Validators;
+
+public sealed class CheckInAttendanceCommandValidator : AbstractValidator<CheckInAttendanceCommand>
+{
+    public CheckInAttendanceCommandValidator()
+    {
+        RuleFor(a => a.Attendance.EmployeeId)
+            .NotEmpty()
+            .WithMessage("Employee ID is required.");
+
+        RuleFor(a => a.Attendance.CheckInTime)
+            .NotEmpty()
+            .WithMessage("Check-in time is required.");
+
+        RuleFor(a => a.Attendance.Status)
+            .NotEmpty()
```

Comment:
You can improve clarity and reuse by using `.Length(1, 10)` instead of `NotEmpty()` plus `MaximumLength()` for `Status`. This communicates the valid range more explicitly.

**Suggested change**
```suggestion
Replace:
```csharp
RuleFor(a => a.Attendance.Status)
    .NotEmpty()
    .MaximumLength(10)
    .WithMessage("Status is required and must not exceed 10 characters.");
```
With:
```csharp
RuleFor(a => a.Attendance.Status)
    .Length(1, 10)
    .WithMessage("Status is required and must not exceed 10 characters.");
```
```